src/main/java/org/folio/rest/impl/InvoiceStorageImpl.java

```diff
@@ -158,6 +173,106 @@ public void putInvoiceStorageInvoicesById(String id, String lang, Invoice entity
     });
   }
 
+  @Validate
+  @Override
+  public void getInvoiceStorageInvoicesDocumentsById(String id, String lang, Map<String, String> okapiHeaders,
+      Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+
+    String cqlQuery = INVOICE_ID_FIELD_NAME + "=" + id;
+    PgUtil.get(DOCUMENT_TABLE, DocumentMetadata.class, DocumentCollection.class, cqlQuery, 0, 999, okapiHeaders, vertxContext,
```

Comment:
As an alternative approach `pageable` trait can be used so client can define how many records are required

Reply:
implemented this endpoint to support sort and filtering